packages/apify/src/charging.ts

```diff
@@ -117,12 +135,47 @@ export class ChargingManager {
         return this.pricingModel === 'PAY_PER_EVENT';
     }
 
+    private loadPricingInfo(
+        pricingInfo: ActorRunPricingInfo | undefined,
+        maxTotalChargeUsd: number,
+    ) {
+        this.pricingModel = pricingInfo?.pricingModel;
+
+        // Load per-event pricing information
+        if (pricingInfo?.pricingModel === 'PAY_PER_EVENT') {
+            for (const [eventName, eventPricing] of Object.entries(
+                pricingInfo.pricingPerEvent.actorChargeEvents,
+            )) {
+                this.pricingInfo[eventName] = {
+                    price: eventPricing.eventPriceUsd,
+                    title: eventPricing.eventTitle,
+                };
+            }
+
+            this.maxTotalChargeUsd = maxTotalChargeUsd;
+        }
+    }
+
+    private loadChargedEventCounts(
+        chargedEventCounts: Record<string, number> | undefined,
+    ) {
+        this.chargingState = {};
+
+        for (const [eventName, chargeCount] of Object.entries(
+            chargedEventCounts ?? {},
+        )) {
+            this.chargingState[eventName] = {
+                chargeCount,
+                totalChargedAmount:
+                    chargeCount * (this.pricingInfo[eventName]?.price ?? 0),
```

Comment:
nit: you have to call the new `load*` methods in this specific order, as `loadChargedEventCounts` reads variables initialized by `loadPricingInfo` (temporal coupling)